src/main.cpp

```diff
@@ -307,6 +310,9 @@ void setup()
     xTaskCreatePinnedToCore(readGPSTask, "ReadGPSTask", 3000, NULL, 1, &GPSTaskHandle, 1);
     xTaskCreatePinnedToCore(SDWriteTask, "SDWriteTask", 4000, NULL, 1, &SDWriteTaskHandle, 1);
 
+    reactionWheelParams rollTaskParam = {&mpu, &dmpReady};
+    xTaskCreate(TaskRollControl, "ReactionWheel", 10000, (void *)&rollTaskParam, 1, &TaskRollControl_Handler);
```

Comment:
- Move the `TaskRollControl` function to main and make it as slim as possible. Most of the core functionality can still be done on the `lib/reaction-wheel/reaction-wheel.cpp`.
- Pin the task to core 0 and write the wrapper task to it as it has been done on `void GetDataTask(void *parameter)` function
- Just a question how long does the `TaskRollControl` task take to complete? If it is under 50ms you can add it to the `GetDataTask` if not consider adding it to core 1

Reply:
on the first point it's not advisable just from the point of readability and the number of variables involved, it would make maintaining the code messy

Reply:
moved task roll control to main